plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -2069,6 +2072,74 @@ public void testSplitPruningForFilterOnPartitionColumn()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test(dataProvider = "testDataMappingSmokeTestDataProvider")
+    @Flaky(issue = "https://github.com/trinodb/trino/issues/5172", match = "Couldn't find operator summary, probably due to query statistic collection error")
+    public void testSplitPruningForFilterOnNonPartitionColumn(DataMappingTestSetup testSetup)
+    {
+        if (testSetup.isUnsupportedType()) {
+            return;
+        }
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_split_pruning_non_partitioned", "(col " + testSetup.getTrinoTypeName() + ")")) {
+            String tableName = table.getName();
+            // Insert separately to ensure two files with one value each
+            assertUpdate("INSERT INTO " + tableName + " VALUES " + testSetup.getSampleValueLiteral(), 1);
+            assertUpdate("INSERT INTO " + tableName + " VALUES " + testSetup.getHighValueLiteral(), 1);
+
+            int splitCount = supportsIcebergFileStatistics(testSetup.getTrinoTypeName()) ? 1 : 2;
+            verifySplitCount("SELECT * FROM " + tableName, 2);
+            verifySplitCount("SELECT * FROM " + tableName + " WHERE col = " + testSetup.getSampleValueLiteral(), splitCount);
+            verifySplitCount("SELECT * FROM " + tableName + " WHERE col > " + testSetup.getSampleValueLiteral(), splitCount);
+            verifySplitCount("SELECT * FROM " + tableName + " WHERE col < " + testSetup.getHighValueLiteral(), splitCount);
+        }
+    }
+
+    boolean supportsIcebergFileStatistics(String typeName)
+    {
+        return !(typeName.equalsIgnoreCase("boolean") ||
+                typeName.equalsIgnoreCase("varbinary") ||
+                typeName.contains("timestamp"));
```

Comment:
Should we have issues for these? timestamp would be especially useful to have

Reply:
Timestamp would have been implemented by https://github.com/trinodb/trino/pull/4526 if I had wrapped it up. I'll resolve the conflicts and reopen it.